### PR TITLE
Remove hard-coded indices from pallet-xcm tests

### PR DESCRIPTION
**Pull request by chungquantin**
# ISSUE
- Link to issue: https://github.com/paritytech/polkadot-sdk/issues/4237

# DESCRIPTION
Remove all ModuleError with hard-coded indices to pallet Error. For example:
```rs
Err(DispatchError::Module(ModuleError {
	index: 4,
	error: [2, 0, 0, 0],
	message: Some("Filtered")
}))
```
To 
```rs
let expected_result = Err(crate::Error::<Test>::Filtered.into());
assert_eq!(result, expected_result);
```
# TEST OUTCOME
```
test result: ok. 74 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s
```

### CODE REVIEW

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs [hunk 1]
```diff
@@ -907,7 +869,7 @@ fn destination_asset_reserve_and_destination_fee_reserve_call<Call>(
 		assert_eq!(result, expected_result);
 		if expected_result.is_err() {
 			// short-circuit here for tests where we expect failure
-			return
+			return;
```
Comment (ggwpez) [inline polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs:872]: Please no unrelated changes.

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs [hunk 2]
```diff
@@ -1632,7 +1550,7 @@ fn local_asset_reserve_and_teleported_fee_call<Call>(
...
```
Comment (ggwpez) [inline polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs:1553]:
```suggestion
			return
```

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs [hunk 3]
```diff
@@ -2088,14 +1978,7 @@ fn reserve_transfer_assets_with_teleportable_asset_disallowed() {
...
 		);
-		assert_eq!(
-			res,
-			Err(DispatchError::Module(ModuleError {
-				index: 4,
-				error: [2, 0, 0, 0],
-				message: Some("Filtered")
-			}))
-		);
+		assert_eq!(res, Err(crate::Error::<Test>::Filtered.into()));
```
Comment (ggwpez) [inline polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs:1981]: There is also `assert_err!`

**Review by ggwpez — APPROVED** [review]
Thanks! Looks better now.

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs (hunk 3 above)
Comment (chungquantin) [reply polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs]: @ggwpez  Resolved in this commit https://github.com/paritytech/polkadot-sdk/pull/4248/commits/9d552711d9a2c95be9478786bf5665a4113be639. Thank you for feedback!

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs (hunk 1 above)
Comment (chungquantin) [reply polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs]: @ggwpez  Resolved in this commit https://github.com/paritytech/polkadot-sdk/pull/4248/commits/9d552711d9a2c95be9478786bf5665a4113be639. Thank you for feedback!

**Review by acatangiu — APPROVED** [review]

**Comment by acatangiu** [comment]
bot fmt

**Comment by acatangiu** [comment]
@chungquantin the bot is not allowed to push to your fork, you'll have to `cargo +nightly fmt --all` yourself and push the changes.
